update context checks

muhammad-ammar opened the pull request:
Update context checks for implicit and explicit access for all resources access.

muhammad-ammar commented on edx_rbac/utils.py:
```diff
@@ -89,6 +93,9 @@ def request_user_has_implicit_access_via_jwt(decoded_jwt, role_name, context=Non
             return True
         else:
             return feature_roles[role_name] == context
+    elif role_name in system_wide_roles and context == ALL_ACCESS_CONTEXT:
```
This is to handle the access for `enterprise_openedx_operator` role. operator roles don't have context attached to them. Please [see](https://github.com/edx/edx-enterprise-data/pull/116/files#r271807400)

muhammad-ammar replied:
Also for the `enterprise_openedx_operator` role `if role_name in feature_roles:` will be not executed.

brittneyexline replied:
the enterprise_openedx_operator role should also map to feature specific roles, but would have a context of `*`, so that we are only checking feature specific roles, which is the purpose of having the mapping.

muhammad-ammar commented:
@brittneyexline I have addressed your feedback. Please have a look. I have also created https://github.com/edx/edx-enterprise/pull/450 to update the context for `enterprise_openedx_operator` role.

brittneyexline commented on tests/test_utils.py:
```diff
@@ -138,6 +139,23 @@ def test_request_user_has_implicit_access_via_jwt_with_context(self):
+        assert request_user_has_implicit_access_via_jwt(
+            toy_decoded_jwt,
+            'enterprise_data_admin',
+            ALL_ACCESS_CONTEXT
```
i think we'd want to pass in a normal context here, no?

brittneyexline commented on tests/test_utils.py:
```diff
@@ -224,6 +242,24 @@ def test_user_has_access_via_database_with_context(self):
+                self.user,
+                'coupon-manager',
+                ConcreteUserRoleAssignment,
+                ALL_ACCESS_CONTEXT
```
same here - the idea is we ask: do you have this role with this context? and the answer is yes if the user's role context matches exactly or is `*`

muhammad-ammar commented:
@brittneyexline I have updated the tests and going to merge this. If still there is something left, let me know and I will address that in a future PR.
